etcdserver/apply.go

```diff
@@ -299,6 +310,9 @@ func (a *applierV3backend) Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error) {
 	if err := a.checkRequestRange(reqs); err != nil {
 		return nil, err
 	}
+	if err := a.checkRequestPermission(header, reqs); err != nil {
```

Comment:
let's check permission in the first phase. we probably should return permission denied, even if we are going to execute success path but there are permission issues in failure path.